example/ssd/dataset/iterator.py

```diff
@@ -159,32 +161,32 @@ def _data_augmentation(self, data, label):
                 xmax = int(crop[2] * width)
                 ymax = int(crop[3] * height)
                 if xmin >= 0 and ymin >= 0 and xmax <= width and ymax <= height:
-                    data = data[ymin:ymax, xmin:xmax, :]
+                    data = mx.img.fixed_crop(data, xmin, ymin, xmax-xmin, ymax-ymin)
                 else:
                     # padding mode
                     new_width = xmax - xmin
                     new_height = ymax - ymin
                     offset_x = 0 - xmin
                     offset_y = 0 - ymin
                     data_bak = data
-                    data = np.full((new_height, new_width, 3), 128.)
+                    data = mx.nd.full((new_height, new_width, 3), 128, dtype='uint8')
                     data[offset_y:offset_y+height, offset_x:offset_x + width, :] = data_bak
                 label = rand_crops[index][1]
-
-        if self.is_train and self._rand_mirror:
-            if np.random.uniform(0, 1) > 0.5:
-                data = cv2.flip(data, 1)
-                valid_mask = np.where(label[:, 0] > -1)[0]
-                tmp = 1.0 - label[valid_mask, 1]
-                label[valid_mask, 1] = 1.0 - label[valid_mask, 3]
-                label[valid_mask, 3] = tmp
-
         if self.is_train:
             interp_methods = [cv2.INTER_LINEAR, cv2.INTER_CUBIC, cv2.INTER_AREA, \
                               cv2.INTER_NEAREST, cv2.INTER_LANCZOS4]
         else:
             interp_methods = [cv2.INTER_LINEAR]
         interp_method = interp_methods[int(np.random.uniform(0, 1) * len(interp_methods))]
-        data = resize(data, self._data_shape, interp_method)
-        data = transform(data, self._mean_pixels)
+        data = mx.img.imresize(data, self._data_shape[0], self._data_shape[1], interp_method)
+        if self.is_train and self._rand_mirror:
+            if np.random.uniform(0, 1) > 0.5:
+                data = mx.nd.flip(data, axis=1)
+                valid_mask = np.where(label[:, 0] > -1)[0]
+                tmp = 1.0 - label[valid_mask, 1]
+                label[valid_mask, 1] = 1.0 - label[valid_mask, 3]
+                label[valid_mask, 3] = tmp
+        data = mx.nd.transpose(data, (2,0,1))
```

Comment:
I profiling this. There is no noticeable performance diff. If you insist, I can push the change